crates/matrix-sdk/src/event_cache/linked_chunk/mod.rs

```diff
@@ -888,11 +888,7 @@ impl<'a, const CAP: usize, Item, Gap> Iterator for IterBackward<'a, CAP, Item, G
     type Item = &'a Chunk<CAP, Item, Gap>;
 
     fn next(&mut self) -> Option<Self::Item> {
-        self.chunk.map(|chunk| {
-            self.chunk = chunk.previous();
-
-            chunk
-        })
+        self.chunk.inspect(|chunk| self.chunk = chunk.previous())
```

Comment:
Changes in this file seem unrelated?

Reply:
That commit was part of #3958, why didn't github notice :shrug:, it did change the base from that branch to main automagically.

Reply:
Rebased locally and force pushed to get rid of the extra commit.